.github/workflows/CD.yml

```diff
@@ -0,0 +1,119 @@
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+          fetch-depth: 0
+      
+      - name: Check no jumps in tags
+        run: |
+          # The format of a tag (version) should be MAJOR.MINOR
+          # If a previous tag is MAJOR.MINOR, valid tags are only MAJOR.(MINOR+1) or (MAJOR+1).0
+          if [[ $(git tag -l | wc -l) != 1 ]]; then # If there is only one tag, no need to check
+            # Find the previous tag
+            prev=$(git tag -l --sort=-v:refname | head -n 2 | tail -n 1)
+            # Split the previous tag into an array of MAJOR and MINOR version components. E.g., for `3.11` it returns the array (3 11)
+            ver_comp=($(tr "." " " <<< $prev))
+            val1="${ver_comp[0]}.$((${ver_comp[1]}+1))" # MAJOR.(MINOR+1)
+            val2="$((${ver_comp[0]}+1)).0" # (MAJOR+1).0
+            if [[ "${{github.ref_name}}" != "$val1" ]] && \
+            [[ "${{github.ref_name}}" != "$val2" ]]; then
+                echo "Invalid tag \"${{github.ref_name}}\" when previous tag is \"$prev\"."
+                exit 1
+            fi
+          fi
+  
+  get-package-name:
+    name: Get package name
+    runs-on: ubuntu-latest
+    needs: validate-tag
+    outputs:
+      package-name: ${{ steps.get-package-name.outputs.package-name }}
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+          fetch-depth: 0
+      
+      - name: Get name
+        id: get-package-name
+        run: |
+          echo "package-name=$(yq '.project.name' pyproject.toml)" >> $GITHUB_OUTPUT
+  
+  release-conda-package:
+    name: Build with conda and release
+    runs-on: ubuntu-latest
+    needs: [get-package-name, validate-tag]
+    env:
+      BUILD_FOLDER: ${{github.workspace}}/build
+      PACKAGE_PATH: ${{github.workspace}}/build/noarch/*.tar.bz2
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+            fetch-tags: true
+            fetch-depth: 0 
+
+      - name: Setup conda build environment
+        uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3.0.4
+        with:
+          miniconda-version: "latest"
+          python-version: ${{ vars.PY_VERSION }} 
+          environment-file: .conda/env_build.yml
+          auto-activate-base: false
+          auto-update-conda: false
+          show-channel-urls: true
+      
+      - name: Build conda package
+        shell: bash -el {0}
+        run: |
+          conda build . --no-anaconda-upload --output-folder=${{env.BUILD_FOLDER}} -c conda-forge -c accessnri -c coecms
+      
+      - name: Upload conda package to Anaconda.org
+        shell: bash -el {0}
+        run: |
+          anaconda -t ${{ secrets.ANACONDA_TOKEN }} upload --user ${{ secrets.ANACONDA_USER_NAME }} ${{env.PACKAGE_PATH}}
```

Comment:
This is fine, but in the future it would be good to standardise on a workflow for this like [uibcdf/action-build-and-upload-conda-packages](https://github.com/uibcdf/action-build-and-upload-conda-packages).

We're using it in [`payu`](https://github.com/payu-org/payu/blob/master/.github/workflows/CD.yml#L67-L74) and [intake-accessnri-catalogue](https://github.com/ACCESS-NRI/access-nri-intake-catalog/blob/main/.github/workflows/cd.yml#L54-L61).

Reply:
I know, but that action doesn't allow to preserve the build file(s) as an artifact, as I need here.
That's why I build and upload to `anaconda.org` manually.

If you think we should generalise this in an action, I can create it under "https://github.com/ACCESS-NRI/actions".

Reply:
> If you think we should generalise this in an action, I am happy to create it under "https://github.com/ACCESS-NRI/actions".

It is worth making an issue and discuss where it could be re-used and make sure it is worth the effort, and that the design will capture as many as possible use-cases.

We could contribute the changes back to the uibcdf action. They seemed pretty happy to collaborate.

But fine to keep as-is for this PR.